util/etcd/etcd.go

```diff
@@ -0,0 +1,368 @@
+
+// Client provides connection parameters for an etcd cluster
+type Client struct {
+	log           logr.Logger
+	timeout       time.Duration
+	backoffParams wait.Backoff
+	endpoint      string
+	etcdClient    *clientv3.Client
+}
+
+// MemberAlarm represents an alarm type association with a cluster member
+type MemberAlarm struct {
+	// MemberID is the ID of the member associated with the raised alarm
+	MemberID uint64
+
+	// Type is the type of alarm which has been raised.
+	Type AlarmType
+}
+
+type AlarmType int32
+
+const (
+	// AlarmOK denotes that the cluster member is OK
+	AlarmOk AlarmType = iota
+	// AlarmNoSpace denotes that the cluster member has run out of disk space
+	AlarmNoSpace AlarmType = iota
+	// AlarmCorrupt denotes that the cluster member has corrupted data
+	AlarmCorrupt AlarmType = iota
+)
+
+// Adapted from kubeadm
+
+// Member struct defines an etcd member; it is used to avoid spreading
+// github.com/coreos/etcd dependencies
+type Member struct {
+	// ID is the ID of this cluster member
+	ID uint64
+	// Name is the human-readable name of the member. If the member is not started, the name will be an empty string.
+	Name string
+	// PeerURLs is the list of URLs the member exposes to the cluster for communication.
+	PeerURLs []string
+	// ClientURLs is the list of URLs the member exposes to clients for communication. If the member is not started, clientURLs will be empty.
+	ClientURLs []string
+	// IsLearner indicates if the member is raft learner.
+	IsLearner bool
+	// Alarms is the list of alarms for a member
+	Alarms []AlarmType
+}
+
+type Status struct {
+	// Leader is the member ID which the responding member believes is the current leader.
+	Leader uint64
+
+	// Version is the cluster protocol version used by the responding member
+	Version string
+
+	// Errors contains alarm/health information and status.
+	Errors []string
+
+	// IsLearner indicates if the member is raft learner.
+	IsLearner bool
+
+	// RaftIndex is the current raft committed index of the responding member.
+	RaftIndex uint64
+
+	// RaftTerm is the current raft term of the responding member.
+	RaftTerm uint64
+
+	// RaftAppliedIndex is the current raft applied index of the responding member.
+	RaftAppliedIndex uint64
+
+	// DbSizeInUse is the size of the backend database logically in use, in bytes, of the responding member.
+	DbSizeInUse int64
+
+	// DdSize is the size of the backend database physically allocated, in bytes, of the responding member.
+	DbSize int64
+}
+
+// New creates a new etcd client with a custom dialer
+func NewClient(dialer grpcDial, endpoint string, tlsConfig *tls.Config, logger logr.Logger) (*Client, error) {
+	if endpoint == "" {
+		endpoint = "localhost"
+	}
+	etcdClient, err := clientv3.New(clientv3.Config{
+		Endpoints:   []string{endpoint},
+		DialTimeout: etcdTimeout,
+		DialOptions: []grpc.DialOption{
+			grpc.WithBlock(), // block until the underlying connection is up
```

Comment:
```suggestion
			grpc.WithBlock(), // Block until the underlying connection is up.
```

Consistency with the other comments and the rest of the repository.